src/iuse_actor.cpp

```diff
@@ -2406,10 +2413,19 @@ void cast_spell_actor::info( const item &, std::vector<iteminfo> &dump ) const
     }
 }
 
-int cast_spell_actor::use( player &p, item &itm, bool, const tripoint & ) const
+int cast_spell_actor::use( player &p, item &it, bool, const tripoint & ) const
 {
+    if( p.has_item( it ) && need_worn && !p.is_worn( it ) ) {
```

Comment:
The check for `has_item` seems to be redundant as you check for `is_worn` as well. If the character doesn't have the item, it *can* not be worn. A check for `need_worn && !is_worn` should be enough.

Same for the checks above and below and for the `is_wielding` check.

Reply:
Thanks. I forgot to remove that when rebasing.